### PR TITLE
feat: add custom image (include ofed installation)

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: user-data.sh [hunk 1]
```diff
@@ -14,6 +14,9 @@ EOL
 cp -R /home/${user}/.ssh/* /root/.ssh/
 chown -R ${user}:${user} /home/${user}/.ssh/
 
+while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
```
Comment (assafgi) [inline user-data.sh:17]: No need for sudo.
Why do you need it?

File: user-data.sh [hunk 2]
```diff
@@ -26,23 +29,27 @@ INSTALLATION_PATH="/tmp/weka"
...
+  cd $INSTALLATION_PATH/$OFED_NAME/*/
+  ./mlnxofedinstall --without-fw-update --add-kernel-support --force 2>&1 | tee /tmp/weka_ofed_installation
+  /etc/init.d/openibd restart
+
+  disk=$(lsblk -o NAME,HCTL,SIZE,MOUNTPOINT | grep "3:0:0:0" | awk '{print $1}')
+  apt update -y
+  apt install -y jq
 else
-  wget http://content.mellanox.com/ofed/MLNX_OFED-${ofed_version}/MLNX_OFED_LINUX-${ofed_version}-ubuntu18.04-x86_64.tgz -O $INSTALLATION_PATH/$OFED_NAME.tgz
+  disk=$(lsblk -o NAME,HCTL,SIZE,MOUNTPOINT | grep "1:0:0:0" | awk '{print $1}')
```
Comment (assafgi) [inline user-data.sh:49]: no need for else, just put this line out of the if instead of duplicating it.

File: variables.tf [hunk 3]
```diff
@@ -42,14 +42,19 @@ variable "cluster_size" {
...
+}
+
+variable "default_ubuntu_vm_id" {
+  type        = map(string)
+  description = "The default azure vm image reference."
   default     = {
     publisher = "Canonical"
     offer     = "UbuntuServer"
-    sku       = "18.04-LTS"
+    sku       = "18_04-lts-gen2"
```
Comment (assafgi) [inline variables.tf:57]: why did you change?

File: vmss.tf [hunk 4]
```diff
@@ -73,7 +74,100 @@ resource "azurerm_proximity_placement_group" "ppg" {
...
-resource "azurerm_linux_virtual_machine_scale_set" "vmss" {
+data "azurerm_shared_image_version" "shared_image_version" {
+ count               = length(var.custom_image) > 0 ? 1 : 0
+ gallery_name        = lookup(var.custom_image, "gallery_name", null)
+ image_name          = lookup(var.custom_image, "definition_name", null)
+ name                = lookup(var.custom_image, "version", null)
+ resource_group_name = lookup(var.custom_image, "rg_name", null)
+}
+
+data azurerm_platform_image "image" {
```
Comment (assafgi) [inline vmss.tf:85]: you don't really need it, you anyway do lookup, so why not just do it on the vmss instead of having this data.

File: vmss.tf [hunk 5]
```diff
@@ -73,7 +74,100 @@ resource "azurerm_proximity_placement_group" "ppg" {
   tags                = merge(var.tags_map, {"weka_cluster": var.cluster_name})
 }
 
-resource "azurerm_linux_virtual_machine_scale_set" "vmss" {
+data "azurerm_shared_image_version" "shared_image_version" {
```
Comment (assafgi) [inline vmss.tf:77]: eventually you just need the imager id, so this should be the parameter, no need for all the variables and no need for this data resource

File: user-data.sh (hunk 1 above)
Comment (denise89p) [reply user-data.sh:17]: I will remove sudo, but I need it, its failed on apt lock

File: variables.tf (hunk 3 above)
Comment (denise89p) [reply variables.tf:57]: its a latest version of ubuntu 18.04

File: user-data.sh (hunk 2 above)
Comment (assafgi) [reply user-data.sh:49]: you do this line in both cases, it should be out of if, this is why no need for else

File: vmss.tf (hunk 4 above)
Comment (denise89p) [reply vmss.tf:85]: right, but if AZURE will give me answer maybe we will need it for custom image and default image, but for now you are right!

File: vmss.tf (hunk 5 above)
Comment (denise89p) [reply vmss.tf:77]: I am using shared gallery want custom image to be public so i need **shared_image_version** and you are right I can use id but I dont like it,` ` **"id": "/Subscriptions/d2f248b9-d054-477f-b7e8-413921532c2a/Providers/Microsoft.Compute/Locations/eastus/Publishers/zscaler1579058425289",`**`

Comment (assafgi) [reply vmss.tf:77]: but it is better then 4 params in a map in tfvars ..

Comment (denise89p) [reply vmss.tf:77]: OK I will change it
